chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,253 @@
+        cov = x_hat.dot(x_hat.T) / x.dtype.type(m) + I
+        self.eigvals, self.eigvectors = xp.linalg.eigh(cov)
+        U = xp.diag(self.eigvals ** -0.5).dot(self.eigvectors.T)
+        self.y_hat_pca = U.dot(x_hat)  # PCA whitening
+        y_hat = self.eigvectors.dot(self.y_hat_pca)  # ZCA whitening
+
+        y = y_hat.reshape((C, b * g,) + x.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if self.groups > 1:
+            y = y.reshape((-1, c, ) + x.shape[2:])
+
+        # Update running statistics
+        q = x.size // C
+        adjust = q / max(q - 1., 1.)  # unbiased estimation
+        self.expected_mean *= self.decay
+        self.expected_mean += (1 - self.decay) * mean
+        self.expected_projection *= self.decay
+        projection = self.eigvectors.dot(U)
+        self.expected_projection += (1 - self.decay) * adjust * projection
+
+        return y,
+
+    def backward(self, indexes, grad_outputs):
+        gy, = grad_outputs
+
+        f = DecorrelatedBatchNormalizationGrad(
+            self.groups, self.eigvals, self.eigvectors, self.y_hat_pca)
+        return f(gy),
+
+
+class DecorrelatedBatchNormalizationGrad(function.Function):
+
+    def __init__(self, groups, eigvals, eigvectors, y_hat_pca):
+        self.groups = groups
+        self.eigvals = eigvals
+        self.eigvectors = eigvectors
+        self.y_hat_pca = y_hat_pca
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        gy = inputs[0]
+        xp = cuda.get_array_module(gy)
+        spatial_ndim = len(gy.shape[2:])
+        spatial_axis = tuple(range(2, 2 + spatial_ndim))
+        b, c = gy.shape[:2]
+        g = self.groups
+        C = c // g
+        m = b * g
+        for i in spatial_axis:
+            m *= gy.shape[i]
+
+        if self.groups > 1:
+            gy = gy.reshape((b * g, C, ) + gy.shape[2:])
+        gy_hat = gy.transpose((1, 0) + spatial_axis).reshape((C, -1))
+
+        gy_hat_pca = self.eigvectors.T.dot(gy_hat)
+        f = gy_hat_pca.mean(axis=1)
+
+        K = self.eigvals[:, None] - self.eigvals[None, :]
+        valid = K != 0
+        K[valid] = 1 / K[valid]
+        xp.fill_diagonal(K, 0)
+
+        V = xp.diag(self.eigvals)
+        V_sqrt = xp.diag(self.eigvals ** 0.5)
+        V_invsqrt = xp.diag(self.eigvals ** -0.5)
+
+        F_c = gy_hat_pca.dot(self.y_hat_pca.T) / gy.dtype.type(m)
+        M = xp.diag(xp.diag(F_c))
+
+        S = 2 * _sym(K.T * (V.dot(F_c.T) + V_sqrt.dot(F_c).dot(V_sqrt)))
+        R = gy_hat_pca - f[:, None] + (S - M).T.dot(self.y_hat_pca)
+        gx_hat = R.T.dot(V_invsqrt).dot(self.eigvectors.T).T
+
+        gx = gx_hat.reshape((C, b * g,) + gy.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if self.groups > 1:
+            gx = gx.reshape((-1, c, ) + gy.shape[2:])
+
+        self.retain_outputs(())
+        return gx,
+
+    def backward(self, inputs, grad_outputs):
+        raise NotImplementedError
```

Comment:
Could you add short TODO comment, preferably explaining why it's not yet implemented?